components/802.15.4_RF/stm-s2lp-rf-driver/source/at24mac_s2lp.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2019-2019 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "NanostackRfPhys2lp.h"
+#include "at24mac_s2lp.h"
+
+
+#if DEVICE_I2C
+#ifdef AT24MAC
```

Comment:
Is this copy&paste from Atmel driver, and why was it not working for this device?

Reply:
It is a copy and it would work with this device. But I guess we cannot rely we always have atmel drivers compiled when running s2-lp? We should probably have some common driver folder for this.

Reply:
Then move to common folder.
AT24MAC is not Atmel RF driver specific. So now would be correct time to separate it.

Reply:
Suggesting to make it as a separate PR once the change has been properly analysed and correct location/API has been decided.

Reply:
A PR for moving AT24MAC: https://jira.arm.com/browse/IOTTHD-3532